woocommerce/woocommerce
# Add new email for notifying customers when an order fails

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #51806



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to `wp-admin/admin.php?page=wc-settings&tab=email` and check there are 2 `Failed order` emails (one for the customer and another for the merchant).
2. Install the `WooCommerce Stripe Gateway` plugin.
3. Check the `Enable Stripe` and `Enable test mode` options in the settings.
4. Go to the store, add a product to the cart, and go to the checkout.
5. Use a [decline test card](https://docs.stripe.com/testing#declined-payments) as a payment method and try to place the order.
6. You should see an error message: `Your card was declined.`
<img width="801" alt="Screenshot 2024-10-01 at 10 14 59" src="https://github.com/user-attachments/assets/638fd107-56b7-4749-9f5f-4ffbb21fbe14">

7. Check you also received a message in the merchant email with the subject: `[store]: Order #XX has failed`. And another message in the customer email with the subject: `Your order at [store] was unsuccessful`.
Check the body of the email matches what's specified in [this comment](https://github.com/woocommerce/woocommerce/pull/51778#issuecomment-2419211553).

| HTML | Plain text |
|--------|--------|
| <img width="500" alt="Screenshot 2024-10-17 at 14 48 56" src="https://github.com/user-attachments/assets/79571ec5-d609-43e9-9ac1-cbae2033be43"> | <img width="528" alt="Screenshot 2024-10-17 at 14 48 30" src="https://github.com/user-attachments/assets/e5b36bff-a414-4e43-8390-c5e95a9f64e3"> | 

8. Go to `wp-admin/admin.php?page=wc-settings&tab=email` and click on the `Failed order` email for the **customer**.
9. Change the `Email type` option to `Plain text` and save.
10. Repeat steps 4-7 and check the message received by the customer looks like the `Plain text` one in the table above.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Send a customer notification email when an order is marked as "failed".
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
